content/blog/2025-12-16-ChocoPy2Bril.md

```diff
@@ -0,0 +1,97 @@
+
+Finally, testing also played an important role in our implementation. We curated a test suite of 20 unit tests that covered the various features we implemented in our ChocoPy2Bril translator. Each test case consisted of a ChocoPy program, and we wrote a script that verified that the result of running the generated Bril code matched the ChocoPy program's output. This helped us ensure the correctness of our implementation and catch any bugs or edge cases. Our test suites were partially adapted/inspired from the test suites provided in the "yangdanny97/chocopy-python-compiler" repository. Initially, we wanted to directly use tests from "yangdanny97/chocopy-python-compiler". However, we soon realized that most of those tests were fairly complex and covered a lot of ChocoPy features that we were not planning on implementing. As a result, we had to edit them to fit our implemented features, as well as manually write some of our tests from scratch. Additionally, on the GitHub issue for this project, we were intially planning on using Turnt for testing. However, we soon realized that since the way that we were checking correctness/calling functions differed across test cases, it was simply easier to write our own testing script that handled each test case individually.
+
+This is an example of one of our test cases, `is_even`:
+```ChocoPy
+def is_even(n: int) -> bool:
+    half = n // 2
+    return n == half * 2
+
+x: int = 8
+y: int = 5
+
+is_even_x: bool = False
+is_even_y: bool = False
+is_even_x = is_even(x)
+is_even_y = is_even(y)
+print(is_even_x)
+print(is_even_y)
+```
+
+When we run our ChocoPy2Bril translator on this test case, it generates the following Bril code:
```

Comment:
I don't think it's very helpful to include the full Bril listing inline, unless there's something specific you want readers to look at in this listing (in which case you can say that).